packages/builder-vite/source-loader-plugin.ts

```diff
@@ -3,7 +3,7 @@ import sourceLoaderTransform from '@storybook/source-loader';
 import type { ExtendedOptions } from './types';
 import MagicString from 'magic-string';
 
-const storyPattern = /\.stories\.[jt]sx?$/;
+const storyPattern = /\.stories\.(([jt]sx?)|(vue))$/;
```

Comment:
If we change this storyPattern, we'll run the `sourceLoaderTransform` on .vue files, right?  IIUC, this is relying on your plugin running first, to convert the file from vue syntax to normal js (csf) syntax.  I wonder, could your addon / vite plugin change the `id` of the file, perhaps in the "resolveId" hook?  In that way, it might be possible to avoid bugs that could happen if someone writes a `.stories.vue` file without your addon.  Though, in that case it would just fail a different way, so maybe it doesn't matter.

Reply:
I tried to change the resolveId, also because it would make a few things easier in the plugin, but never got it to work. If I remember correctly, one of the issues was that the vue loader would always overwrite the resolve id again.

But as you say, it probably doesn't matter that much since it would only change the place where things are failing.